### PR TITLE
speed up solver by skipping timestamp minimization when not needed

### PR DESCRIPTION
**Pull request by jjhelmus**
Speed up the solver by skipping the timestamp minimization step when there is a single "simple" solution after the package count minimization step.

It may be possible to skip other minimization steps using similar logic but further testing is needed.

### CODE REVIEW

File: conda/resolve.py [hunk 1]
```diff
@@ -849,6 +849,19 @@ def mysat(specs, add_if=False):
             constraints = r2.generate_spec_constraints(C, specs)
             return C.sat(constraints, add_if)
 
+        # Return a solution of packages
+        def clean(sol):
+            return [q for q in (C.from_index(s) for s in sol)
+                    if q and q[0] != '!' and '@' not in q]
+
+        # Determine if there are multiple "simple" solutions
```
Comment (soapy1) [inline conda/resolve.py]: Can you expand on what a "simple" solution is

Comment (jjhelmus) [reply conda/resolve.py]: 'simple' or 'obvious' solutions are defined as solution where none of the packages in
the current best solution are used.  I should add this to the comment.

**Comment by jjhelmus** [comment]
Note that this is against the `4.5.x` branch.  

This change can knock some significant time off solve time.  For example the wall time for 
`conda create -nt  --dry-run anaconda` on linux-64 without this change is 270 seconds, with the change is 195 seconds.

File: conda/resolve.py [hunk 2]
```diff
@@ -923,19 +936,18 @@ def mysat(specs, add_if=False):
...
         # Prune unnecessary packages
         eq_c = r2.generate_package_count(C, specm)
         solution, obj7 = C.minimize(eq_c, solution, trymax=True)
         log.debug('Weak dependency count: %d', obj7)
 
-        def clean(sol):
-            return [q for q in (C.from_index(s) for s in sol)
-                    if q and q[0] != '!' and '@' not in q]
+        multiple_solutions = has_multiple_simple_solutions(solution)
+        if multiple_solutions:
```
Comment (soapy1) [inline conda/resolve.py]: Is there an easy way to show how this speeds up a solve? Just curious

Comment (jjhelmus) [reply conda/resolve.py]: If `multiple_solutions` is False (i.e. there is a single solution) then the block of code is not executed.  That block, particularly the line `solution, obj6t = C.minimize(eq_t, solution)` line can take considerable time if a large number of packages are being considered.

**Comment by jjhelmus** [comment]
@soapy1 I tried to clarify the logic being applied to conditionally skip the timestep minimization in 816f344.  Does the new flow and documentation help explain this behavior?

**Comment by soapy1** [comment]
@jjhelmus looks great!

**Comment by msarahan** [comment]
Good addition.  Thanks @jjhelmus

**Pull request merged**